New enable_sql_profiler INI setting for Tracker

simivar opened the pull request:
#14219

Commit:
Add enable_sql_profiler setting to Tracker

Findus23 commented:
I feel like this might get really confusing with the existing `enable_sql_profiler` setting:

https://github.com/matomo-org/matomo/blob/f6908294d00093234b2a27342f8f32b13a2b2232/config/global.ini.php#L142-L145

simivar commented:
That was @tsteur proposition from #14219. I feel like it can be confusing too but do you have any idea how it should be name? I can't think of anything myself.

tsteur commented:
@Findus23 using the sql profiler in tracking you don't want to use when you only want to use the SQL profiler for archiving or web.

tsteur commented on core/Tracker/Response.php:
```diff
@@ -25,7 +25,7 @@ public function init(Tracker $tracker)
         ob_start(); // we use ob_start only because of Common::printDebug, we should actually not really use ob_start
 
-        if ($tracker->isDebugModeEnabled()) {
+        if ($tracker->isDebugModeEnabled() && TrackerConfig::getConfigValue('enable_sql_profiler')) {
```
@simivar I'm wondering maybe it can be useful to enable `enable_sql_profiler` without enabling `$tracker->isDebugModeEnabled()`? Not sure... If we make it independent, we would also need to adjust the code in line `90` where we print the profiling results. I suppose it's fine though. Maybe let's just leave it like this for now.

tsteur committed:
Adjust message

@simivar fyi removed the need for `[log]` as it should be 99% enabled. It's just important they also enable tracker debug.

tsteur commented:
Thanks for this @simivar Also created https://github.com/matomo-org/developer-documentation/pull/302/files to adjust the docs.
